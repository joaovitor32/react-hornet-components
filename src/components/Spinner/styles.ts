import styled, { css } from 'styled-components';

import { Palette, Dimension, SpinnerProps } from '../../types';

type SpinnerType = SpinnerProps<Palette, Dimension>;

const Container = styled.div<SpinnerType>`
  display: flex;
  justify-content: center;
  align-items: center;
  text-align: center;

  position: relative;

  ${(props) =>
    css`
      width: ${props.size.width};
      height: ${props.size.height};
      background-color: ${props.palette.primary};
    `};

  & > div {
    box-sizing: border-box;
    display: block;
    position: absolute;
    width: 80%;
    height: 80%;
    margin: 8px;

    border-radius: 50%;
    animation: lds-ring 1.2s cubic-bezier(0.5, 0, 0.5, 1) infinite;

    ${(props) =>
      !!props.palette &&
      css`
        border: 4px solid ${props.palette.secondary};
        border-color: ${props.palette.secondary} transparent transparent transparent;
      `};

    :nth-child(1) {
      animation-delay: -0.45s;
    }

    :nth-child(2) {
      animation-delay: -0.3s;
    }

    :nth-child(3) {
      animation-delay: -0.15s;
    }

    @keyframes lds-ring {
      0% {
        transform: rotate(0deg);
      }
      100% {
        transform: rotate(360deg);
      }
    }
  }

  ${({ sx }) =>
    !!sx &&
    css`
      ${sx}
    `}
`;

export { Container };
