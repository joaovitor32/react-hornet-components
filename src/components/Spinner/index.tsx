import * as React from 'react';

import { Palette, Dimension, SpinnerProps } from '../../types';

import { Container } from './styles';

const Spinner = ({ palette, size, sx }: SpinnerProps<Palette, Dimension>) => (
  <Container palette={palette} size={size} sx={sx}>
    <div />
    <div />
    <div />
  </Container>
);

export default Spinner;
