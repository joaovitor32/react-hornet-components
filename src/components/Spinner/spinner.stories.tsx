/* eslint-disable no-restricted-syntax */
import React from 'react';

import { Meta, Story } from '@storybook/react';

import Spinner from '.';

export default {
  title: 'Components/Spinner',
  component: Spinner,
} as Meta;

const Template: Story<any> = (args) => <Spinner {...args} />;

export const Primary = Template.bind({});

Primary.args = {
  palette: {
    primary: 'red',
    secondary: 'blue',
  },
  size: {
    width: '80px',
    height: '80px',
  },
};
