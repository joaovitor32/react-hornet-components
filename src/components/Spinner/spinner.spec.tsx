import React from 'react';

import '@testing-library/jest-dom';

import { render, screen } from '@testing-library/react';

import Spinner from '.';

describe('<Spinner />', () => {
  it('should render correctly, --success case', () => {
    const { container } = render(
      <Spinner size={{ width: '', height: '' }} palette={{ primary: '', secondary: '', tertiary: '' }} />
    );

    expect(container.firstChild).toMatchSnapshot();
  });
});
