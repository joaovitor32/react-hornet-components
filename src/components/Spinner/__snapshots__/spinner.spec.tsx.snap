// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Spinner /> should render correctly, --success case 1`] = `
<div
  class="sc-bcXHqe dXZUGq"
>
  <div />
  <div />
  <div />
</div>
`;
